account_invoice_margin_sale/models/account_invoice.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_margin_applicable_lines(self):
+        lines = super()._get_margin_applicable_lines()
+        return lines.filtered(lambda x: not x.sale_line_ids.is_downpayment)
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends("purchase_price", "price_subtotal")
+    def _compute_margin(self):
+        for line in self:
+            if (
+                line.move_id
+                and line.move_id.type[:2] == "in"
+                or any(line.sale_line_ids.mapped("is_downpayment"))
```

Comment:
I didn't see that the other condition is met in `super`, so no need to add it here:
```suggestion
                any(line.sale_line_ids.mapped("is_downpayment"))
```

Reply:
Addressed!